### PR TITLE
Fixing up a few issues in the new LLCP implementation

### PR DESCRIPTION
**Pull request by erbr-ot**
It was found that procedure pause handling was incomplete, so this is completed.
There was a slip in modification to struct llcp - missing member 'is_must_expire'
Finally it has turned out that the tx data pause/resume had a race condition, so the pause flag is changed into a counter, however without changing the API.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c [hunk 1]
```diff
@@ -428,7 +449,7 @@ static void rr_st_idle(struct ll_conn *conn, uint8_t evt, void *param)
 				/* Run remote procedure */
 				rr_act_run(conn);
 				rr_set_state(conn, RR_STATE_TERMINATE);
-			} else if (incompat == INCOMPAT_NO_COLLISION) {
+			} else if (!with_instant || incompat == INCOMPAT_NO_COLLISION) {
```
Comment (thoh-ot) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:452]: Isn't this change unrelated to the procedure pause and belongs in a separate commit regarding procedure collision?

Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:452]: yes it does - thank you!

**Comment by erbr-ot** [comment]
This is still pending reviews

**Review by kruithofa — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]
